### DISCUSSION TITLE
Multisite: identifying if field data is from page or inherited

### DISCUSSION THREAD

**Discussion by sjclark** [statamic/cms#5052] [Troubleshooting] [answered]
I'm trying to figure out if there is a way to **detect if front-end field data has come from the current page or whether it has been inherited from the default locale**.

The usage case is a site I'm developing where they're progressively rolling out other languages but won't necessarily have every page translated yet - in those cases they'd prefer to just serve the English version, but have a little notification ("This page not yet translated, coming soon etc") while staying on the localised page. So it would be handy to be able to detect if the currently viewed page data is inheriting the content from another page.

I found a couple of other issues alluding to this as a limitation; [for example](https://github.com/statamic/cms/issues/3987).

I've tried various `dump`'s to try and locate anything usable, and used the suggestion in the issue above
ie. `{{ if page:fieldname }} or {{ page:fieldname }}` vs `{{ fieldname }}`

But can't seem to find anything usable. Might be handy to have some sort of metadata on the front-end with the data, maybe `#inherited: true` or origin, or something?

Open to other suggestions?

**Comment by jelleroorda** [comment, marked as answer]
I would do this with a custom tag, I haven't tested it very well (might be a bug somewhere), but it should be something like this:

```php
<?php

namespace App\Tags;

use Statamic\Facades\Entry;
use Statamic\Tags\Tags;

class FieldIsTranslated extends Tags
{
    /**
     * The {{ field_is_translated:some_field default="true" }} tag.
     *
     * Finds the ID of the origin for this entry (ID taken from the context)
     * If the entry does not have an origin it is the origin, so we use the id.
     *
     * @return string|array
     */
    public function wildcard($field = false)
    {
        $id = $this->context->get('id', false);
        $defaultValue = $this->params->get('default', 'true') === 'true';

        // If no field was passed, just fall back to the default value
        if (!$field) {
            return $defaultValue;
        }

        // If there is no entry ID, we can't detect if there's a translation.
        if (!$id) {
            return $defaultValue;
        }

        /** @var \Statamic\Entries\Entry $entry */
        $entry = Entry::find($id);

        // If this entry does not have an origin, it is the origin.
        // Origin entries are the 'base translation', so by definition translated.
        if (!$entry->hasOrigin()) {
            return true;
        }

        // Let's compare the field for this translation and the origin entry.
        return $entry->get($field) !== $entry->value($field);
    }
}
```

The usage is as follows:
```html
{{ if {field_is_translated:content} }} 
     The content field was translated.
{{ else }}
     The content field was NOT translated.
{{ /if }}
```

It uses the `id` from the context to find the corresponding entry. Then it compares the specified field between the entry and the origin to see whether they are the same. As you can see from the snippet there are a few edge cases that get handled with a default value, for example if there is no `id` in the context it'll fall back to the default value: `true`. If it can't find the field on the specified entry it'll also fall back to the default value.

So if you want, you should be to change that by using `{{ if {field_is_translated:content default="false"} }}` instead.

Like I said, I haven't tested it very well, but it should get you at least 95% of the way.

**Reply by jasonvarga** [reply]
`$entry->get()` gets the value from that entry only, without inheritance.
`$entry->value()` gets the value _with_ inheritance.

I guess you could do this, too:

```php
if ($entry->get($field) != $entry->value($field)) {
  // it's inherited
}
```

**Reply by jelleroorda** [reply]
True, I think that makes it a little cleaner, I've updated the snippet.

**Reply by sjclark** [reply]
Absolute champion @jelleroorda (and @jasonvarga) 😃 Works great!

Only oddity seems to be on my end it's the exact inverse behaviour, on my end:
```
{{ if {field_is_translated:content} }} 
     The content field was NOT translated.
{{ else }}
     The content field is translated.
{{ /if }}
```
But thats easily fixed!
